src/Controller/UserController.php

```diff
@@ -764,8 +796,36 @@ public function setProjectManagerForUser(
                 'message' => 'Access denied!'
             ], Response::HTTP_FORBIDDEN);
         }
+
+        $data = $request->getContent();
+        /** @var DeserializationContext $context */
+        $context = DeserializationContext::create()->setGroups(array('SetRole'));
+
+        $id = $this->serializer->deserialize(
+            $data,
+            User::class,
+            'json',
+            $context
+        );
+        $projectManager = $userRepository->find($id);
+        if (!$projectManager) {
+            return new JsonResponse(
+                [
+                    'code' => Response::HTTP_NOT_FOUND,
+                    'message' => 'Not found!'
+                ],
+                Response::HTTP_NOT_FOUND
+            );
+        }
+
         if ($projectManager->getRoles() === array('ROLE_USER')) {
-            return new JsonResponse(['message' => 'This user in not a PM!'], Response::HTTP_BAD_REQUEST);
+            return new JsonResponse(
+                [
+                    'code' => Response::HTTP_BAD_REQUEST,
+                    'message' => 'This user in not a PM!'
```

Comment:
The message is bad. It says that the user to be changed is not a PM, not that the current user is not a PM.